src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -599,19 +637,34 @@ private int decodeConnect(
             int reasonCode = SUCCESS;
 
             final MqttConnectFW mqttConnect = mqttConnectRO.tryWrap(buffer, offset, limit);
+            int flags;
             if (mqttConnect == null)
             {
                 reasonCode = PROTOCOL_ERROR;
             }
             else if ((mqttConnect.typeAndFlags() & 0b1111_1111) != CONNECT_FIXED_HEADER ||
-                     (mqttConnect.flags() & 0b0000_0001) != 0b0000_0000)
+                     ((flags = mqttConnect.flags()) & 0b0000_0001) != 0b0000_0000)
```

Comment:
Avoid putting side effects into conditions, it makes the code harder to reason about to verify correctness in all cases.

Instead we normally have a `decode:` label starting the block, with a `break decode;` where needed to escape the block. This makes it easier to have code between blocks, without needing nested `if-else` blocks.